From a0c9b0d3f8ed43430eb97fefdf754b742d68f1dd Mon Sep 17 00:00:00 2001
From: Sergey Dyasli <sergey.dyasli@citrix.com>
Date: Thu, 26 Sep 2019 14:19:54 +0100
Subject: [PATCH] x86/shim: fix ballooning down the guest
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Currently ballooning down a pvshim guest causes the following errors
inside the shim:

    d3v0 failed to reserve 512 extents of order 512 for offlining

And the ballooned-out pages stay inside shim and don't reach L0 Xen.

Fix this by passing the correct arguments to pv_shim_offline_memory()
during a XENMEM_decrease_reservation request.

Signed-off-by: Sergey Dyasli <sergey.dyasli@citrix.com>
diff --git a/xen/common/memory.c b/xen/common/memory.c
index 058014675c..2e2c902a2e 100644
--- a/xen/common/memory.c
+++ b/xen/common/memory.c
@@ -1237,7 +1237,7 @@ long do_memory_op(unsigned long cmd, XEN_GUEST_HANDLE_PARAM(void) arg)
              * hypercall has failed and only part of the extents where
              * processed.
              */
-            pv_shim_offline_memory(args.nr_extents, args.nr_done);
+            pv_shim_offline_memory(args.nr_done, args.extent_order);
 #endif
 
         break;
